code/src/mlx90640_processor.cpp

```diff
@@ -1,107 +1,152 @@
+        // Datasheet section 11.1.3
+        pix_os_ref_c pix_offset(bus, params);
+        // Datasheet section 11.1.4
+        alpha_c alpha(bus, params);
+        // Datasheet section 11.1.5
+        kv_c kv(bus, params);
+        // Datasheet section 11.1.6
+        kta_c kta(bus, params);
 
-    void mlx90640_processor_c::apply_treshold(int &value,
-                                              const uint16_t exceeds,
-                                              const int minus) const {
-        if (value > exceeds) {
-            value -= minus;
-        }
-    }
+        lookupables = {&pix_offset, &alpha, &kv, &kta};
 
-    int mlx90640_processor_c::get_compensated_data(const registers reg_addr,
-                                                   const uint16_t and_bits,
-                                                   const uint8_t shifted,
-                                                   const uint16_t exceeds,
-                                                   const int minus) const {
-        int data = extract_data(reg_addr, and_bits, shifted);
-        apply_treshold(data, exceeds, minus);
-        return data;
-    }
+        for (const auto &table : lookupables) {
+            init_table(*table);
+        }
 
-    /* Dataseheet page 36 */
-    float mlx90640_processor_c::get_Vdd() {
-        Kvdd = get_compensated_data(registers::EE_VDD_PIX, 0xFF00, 8, 127, 256);
-        Kvdd *= 32;
+        // Datasheet section 11.1.1
+        ee_vdd_c ee_vdd(bus, params);
+        // Datasheet section 11.1.2
+        ee_ta_c ee_ta(bus, params);
+        // Datasheet section 11.1.7
+        ee_gain_c ee_gain(bus, params);
+        // Datasheet section 11.1.8
+        ee_ksta_c ksta(bus, params);
+        // Datasheet section 11.1.9
+        ee_corner_temp_c ct(bus, params);
+        // Datasheet section 11.1.10
+        ee_ksto_c ksto(bus, params);
+        // Datasheet section 11.1.11
+        alpha_corr_c alpha_corr(bus, params);
+        // Datasheet section 11.1.12
+        alpha_cp_c alpha_cp(bus, params);
+        // Datasheet section 11.1.13
+        ee_off_cp_c cp_offset(bus, params);
+        // Datasheet section 11.1.14
+        ee_kv_cp_c kv_cp(bus, params);
+        // Datasheet section 11.1.15
+        ee_kta_cp_c kta_cp(bus, params);
+        // Datasheet section 11.1.16
+        ee_tgc_extractor_c tgc(bus, params);
+        // Datasheet section 11.1.17
+        ee_resolution_c resolution(bus, params);
+
+        static_vars = {&ee_vdd, &ee_ta,      &ee_gain,   &ksta,      &ct,
+                       &ksto,   &alpha_corr, &alpha_cp,  &cp_offset, &kv_cp,
+                       &kta_cp, &tgc,        &resolution};
+        // Sets all EEPROM data in mlx params
+        for (const auto &static_var : static_vars) {
+            static_var->extract();
+        }
 
-        Vdd25 = extract_data(registers::EE_VDD_PIX, 0x00FF, 0);
-        Vdd25 = (Vdd25 - 256) * 32 - 8192;
+        hwlib::cout << "Mlx90640 initialized!\n";
```

Comment:
Not required in master

Reply:
Initializing takes some time, I think around 1/2 seconds approxemately, I thought maybe add a print statement to reduce confusion when other people continue to code on this module.